tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -621,7 +621,11 @@ static void spi_loopback_test_word_size(struct spi_dt_spec *spec,
 	struct spi_config config_copy = spec->config;
 
 	config_copy.operation &= ~SPI_WORD_SIZE_MASK;
-	config_copy.operation |= SPI_WORD_SET(word_size);
+	if (word_size == 16) {
+		config_copy.operation |= SPI_WORD_SET(word_size - 1);
+	} else {
+		config_copy.operation |= SPI_WORD_SET(word_size);
+	}
```

Comment:
This commit should be dropped. That may make the STM32 tests pass but it hides the problem.
STM32 supports 16-bit SPI transfer. And `test_spi_word_size_16` function passes when polling or interrupt transfers are used.

The problem lies when we activate the DMA. DMA is configured in dts overlays, and the size is defined there also. So when the DMA is configured for 8-bit transfers, the 16-bit test fails. For the 16-bit test to pass, DMA should be configured as 16-bit, but then the 8-bit test would fail. 

Until recently we relied on `CONFIG_SPI_LOOPBACK_16BITS_FRAMES` to know how to configure the DMA. But with the new tests, it isn't sufficient.

@decsny You introduced the new `test_spi_word_size_16` test. Do you have an idea how to solve this?